File: tests/runtime-settings-test.py
import os
import tempfile
from pathlib import Path

with tempfile.TemporaryDirectory() as temporary:
    os.environ['TEST_BASE_URL'] = 'http://127.0.0.1:8233/'
    os.environ['EXPECTED_PUBLIC_URL'] = 'https://tio2products.com'
    os.environ['EXPECTED_RELEASE'] = '0123456789abcdef0123456789abcdef01234567'
    os.environ['TEST_OUTPUT_DIR'] = temporary
    from support.runtime import runtime_settings

    settings = runtime_settings()
    assert settings['base_url'] == 'http://127.0.0.1:8233'
    assert settings['public_url'] == 'https://tio2products.com/'
    assert settings['release'] == '0123456789abcdef0123456789abcdef01234567'
    assert settings['output_dir'] == str(Path(temporary).resolve())

print('runtime settings contract passed')

File: tests/runtime-settings-test.mjs
import assert from 'node:assert/strict';
import path from 'node:path';

process.env.TEST_BASE_URL='http://127.0.0.1:8233/';
process.env.EXPECTED_PUBLIC_URL='https://tio2products.com';
process.env.EXPECTED_RELEASE='0123456789abcdef0123456789abcdef01234567';
process.env.TEST_OUTPUT_DIR='.runtime/test-results/node-settings';

const {runtime,workspaceContainerPath}=await import('./support/runtime.mjs');
assert.equal(runtime.baseURL,'http://127.0.0.1:8233');
assert.equal(runtime.publicURL,'https://tio2products.com/');
assert.equal(runtime.expectedRelease,'0123456789abcdef0123456789abcdef01234567');
assert.equal(workspaceContainerPath(path.resolve('.runtime/test-results/content-before.json')),'/workspace/.runtime/test-results/content-before.json');
console.log('node runtime settings contract passed');
